lib/ipmi_hpeoem.c

```diff
@@ -0,0 +1,428 @@
+                val2str(rsp->ccode, completion_code_vals));
+        return -1;
+    }
+
+    nic_selection = rsp->data[3];
+    if (nic_selection <= 1)
+    {
+        nic_selection_str = ILONICSelectionMode[nic_selection];
+    }
+    printf("NIC selection: %s\n", nic_selection_str);
+
+    reset_needed = (rsp->data[4] & ILO_CONFIGURATION_STATUS_MASK) == ILO_CONFIGURATION_STATUS_MASK;
+    printf("Reset needed: %s\n", reset_needed ? "yes" : "no");
+
+    // read shared nic information
+    if (nic_selection == ILO_SELECT_NIC_SHARED)
+    {
+        msg_data[1] = IPMI_ILO_SHARED_NIC_SELECTION;
+        rsp = intf->sendrecv(intf, &req);
+        if (!rsp)
+        {
+            lprintf(LOG_ERR, "Error in getting nic selection");
+            return -1;
+        }
+        else if (rsp->ccode)
+        {
+            lprintf(LOG_ERR, "Error in getting nic selection (%s)",
+                    val2str(rsp->ccode, completion_code_vals));
+            return -1;
+        }
+
+        shared_nic_selection = rsp->data[1];
+        if (shared_nic_selection <= 2)
+        {
+            shared_nic_selection_str = ILOSharedNICSelection[shared_nic_selection];
+        }
+        shared_nic_port = rsp->data[2];
+        if (shared_nic_port <= 2)
+        {
+            shared_nic_port_str = ILOSharedNICPort[shared_nic_port];
+        }
+
+        printf("Shared NIC selection: %s\n", shared_nic_selection_str);
+        printf("Shared NIC port: %s\n", shared_nic_port_str);
+
+        printf("Shared NIC support:");
+        if (rsp->data[3] & ILO_SHARED_ALOM_SUPPORTED)
+        {
+            printf(" ALOM");
+        }
+        if (rsp->data[3] & ILO_SHARED_LOM_SUPPORTED)
+        {
+            printf(" LOM");
+        }
+        printf("\n");
+    }
+
+    return 0;
+}
+
+static int
+ipmi_hpeoem_parse_nic_selection_mode(char **argv)
+{
+    if (argv[current_arg] && !strcmp(argv[current_arg], "dedicated"))
+    {
+        return DEDICATED;
+    }
+
+    if (!argv[current_arg] || strcmp(argv[current_arg], "shared"))
+    {
+        return INVALID;
+    }
+    current_arg++;
+    if (!argv[current_arg] || !strcmp(argv[current_arg], "with"))
+    {
+        return INVALID;
+    }
+
+    current_arg++;
+    if (argv[current_arg] && !strcmp(argv[current_arg], "lom1"))
+    {
+        return SHARED_WITH_LOM1;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "lom2"))
+    {
+        return SHARED_WITH_LOM2;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "alom1"))
+    {
+        return SHARED_WITH_ALOM1;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "alom2"))
+    {
+        return SHARED_WITH_ALOM2;
+    }
+    else
+    {
+        return INVALID;
+    }
+    return INVALID;
+}
+
+static int
+ipmi_hpeoem_lan_set_nic_selection(struct ipmi_intf *intf, int channel, int nic_selection)
+{
+    struct ipmi_rs *rsp;
+    struct ipmi_rq req;
+    uint8_t input_length = 0;
+    uint8_t msg_data[10];
```

Comment:
Why `10` if the max you use is `6` ? It looks like `6` is actually the maximum defined for command 224 by iLO manual on page 141.

All in all, please avoid using magic numbers like `10` or `6`. Please use enums or macros with descriptive names.